src/main/java/com/dungeoncrawler/Javiarenka/character/Monster.java

```diff
@@ -23,6 +25,21 @@ public void setDamageStrength(int damageStrength) {
 
     @Override
     void attack(Character hero) {
-        hero.setHp(hero.getHp() - damageStrength);
+        Hero attackedHero = (Hero)hero;
+
+        if (attackedHero.getEquippedArmor().getDamageReduction() <= damageStrength) {
```

Comment:
Super warunek, zapobiega dziwnym sytuacjom, w których jak ktoś dostałby mniej obrażeń niż jego redukcja, to by się nagle uleczył.